python/tests/io/test_pyarrow.py

```diff
@@ -1130,3 +1131,15 @@ def test_projection_filter_on_unknown_field(schema_int_str: Schema, file_int_str
         _ = project(schema, [file_int_str], GreaterThan("unknown_field", "1"), schema_int_str)
 
     assert "Could not find field with name unknown_field, case_sensitive=True" in str(exc_info.value)
+
+
+def test_pyarrow_to_schema_simple(table_schema_simple: Schema, pyarrow_schema_simple: pa.Schema) -> None:
+    actual = str(pyarrow_to_schema(pyarrow_schema_simple))
+    expected = str(table_schema_simple)
+    assert actual == expected
+
+
+def test_pyarrow_to_schema_nested(table_schema_nested: Schema, pyarrow_schema_nested: pa.Schema) -> None:
+    actual = str(pyarrow_to_schema(pyarrow_schema_nested))
```

Comment:
Initially, I was thinking of using `schema_to_pyarrow` and `pyarrow_to_schema` to do a round conversion to verify the correctness. But I found that `schema_to_pyarrow` does not assign field id to the metadata of list element and map key/value.
https://github.com/apache/iceberg/blob/ad0f04dd90cf42011535207e661924c3b0456f1c/python/pyiceberg/io/pyarrow.py#L354-L359
may need to be changed to 
```python
    def list(self, list_type: ListType, element_result: pa.DataType) -> pa.DataType:
        element_field = self.field(list_type.element_field, element_result)
        return pa.list_(value_type=element_field)

    def map(self, map_type: MapType, key_result: pa.DataType, value_result: pa.DataType) -> pa.DataType:
        key_field = self.field(map_type.key_field, key_result)
        value_field = self.field(map_type.value_field, value_result)
        return pa.map_(key_type=key_field, item_type=value_field)
```
to preserve the field id in the metadata

Based on my understanding, such change will only benefit the unit test at current stage. So I choose to not apply the change and hardcode a pyarrow schema fixture in `conftest.py` for testing instead.

Reply:
Thank you for giving this a try. I did some testing on my end, and adding the fields works perfectly fine for Arrow. I've created a table, and read that using PyIceberg:

```sql
CREATE TABLE default.maps_and_lists(
    some_list ARRAY<string>,
    some_amp  MAP<string, int>
)

INSERT INTO default.maps_and_lists VALUES(array("a", "b", "c"), map("a", 1, "b", 2, "c", 3))
```
Wit the changes that you suggested, I can just read it into a dataframe:

```python
def test_vo():
    from pyiceberg.catalog import load_catalog
    cat = load_catalog('local')
    tbl = cat.load_table('default.maps_and_lists')
    df = tbl.scan().to_arrow()
    print(df)
```

Which shows:

```
some_list: list<item: string>
  child 0, item: string
some_amp: map<string, int32>
  child 0, entries: struct<key: string not null, value: int32> not null
      child 0, key: string not null
      child 1, value: int32
----
some_list: [[["a","b","c"]]]
some_amp: [[keys:["a","b","c"]values:[1,2,3]]]
```
I'm in favor of changing this.

Reply:
Thank you for the testing. I applied this change to pyarrow.py